src/aeplugin_dev_mode_handler.erl

```diff
@@ -30,9 +32,28 @@ init(Req, Opts) ->
 content_types_provided(Req, State) ->
     serve_request(Req),
     {[
-       {<<"text/html">>, index_html}
+       {<<"text/html">>, index_html},
+       {<<"application/json">>, json_api}
```

Comment:
I was working on adding `json` support, but the complication here is that `serve_request` is always called first, without grabbing the result. We could stuff the result in the `Req` map, which will then be passed on. I don't know if this would be frowned upon by `cowboy` afficionados, but it ought to work. The `json_api` function could then extract the result, and simply concern itself with formatting it.

Reply:
I was very confused by cowboy first filtering by content types, although it might make sense somewhere. Also it was a bit unexpected seeing the processing of the request being "forked" in two flows: the actual callback function index_html that produces the response has no information about success or failure of serve_request.

Putting the Result from `serve_request` somewhere in the `Req` map is a good idea. That would be helpful for displaying eventual errors or some other form of response (e.g. success - currently, the GUI gives no feedback on sending requests)
 For the JSON API, the way it is right now, I'm handing down `Req` in a waterfall manner, like used to from building `express` backends in NodeJS, so it should be fine.